pkg/diff/diff.go

```diff
@@ -460,21 +447,75 @@ func (t *TableDiff) checkChunkDataEqual(ctx context.Context, filterByRand bool,
 	return equal, nil
 }
 
+// checksumInfo save some information about checksum
+type checksumInfo struct {
+	checksum int64
+	err      error
+	cost     time.Duration
+	tp       string
+}
+
+// check the checksum is equal or not
 func (t *TableDiff) compareChecksum(ctx context.Context, chunk *ChunkRange) (bool, error) {
-	// first check the checksum is equal or not
-	getSourceChecksumTime := time.Now()
-	sourceChecksum, err := t.getSourceTableChecksum(ctx, chunk)
-	if err != nil {
-		return false, errors.Trace(err)
+	ctx1, cancel1 := context.WithCancel(ctx)
+	defer cancel1()
+
+	var (
+		getSourceChecksumDuration, getTargetChecksumDuration time.Duration
+		sourceChecksum, targetChecksum                       int64
+		checksumInfoCh                                       = make(chan checksumInfo)
+		firstErr                                             error
+	)
+	defer close(checksumInfoCh)
+
+	getChecksum := func(db *sql.DB, schema, table, limitRange string, tbInfo *model.TableInfo, args []interface{}, tp string) {
+		beginTime := time.Now()
+		checksum, err := dbutil.GetCRC32Checksum(ctx1, db, schema, table, tbInfo, limitRange, args)
+		cost := time.Since(beginTime)
+
+		checksumInfoCh <- checksumInfo{
+			checksum: checksum,
+			err:      err,
+			cost:     cost,
+			tp:       tp,
+		}
 	}
-	getSourceChecksumDuration := time.Since(getSourceChecksumTime)
 
-	getTargetChecksumTime := time.Now()
-	targetChecksum, err := dbutil.GetCRC32Checksum(ctx, t.TargetTable.Conn, t.TargetTable.Schema, t.TargetTable.Table, t.TargetTable.info, chunk.Where, utils.StringsToInterfaces(chunk.Args))
-	if err != nil {
-		return false, errors.Trace(err)
+	args := utils.StringsToInterfaces(chunk.Args)
+	for _, sourceTable := range t.SourceTables {
+		go getChecksum(sourceTable.Conn, sourceTable.Schema, sourceTable.Table, chunk.Where, t.TargetTable.info, args, "source")
+	}
+
+	go getChecksum(t.TargetTable.Conn, t.TargetTable.Schema, t.TargetTable.Table, chunk.Where, t.TargetTable.info, args, "target")
+
+	for i := 0; i < len(t.SourceTables)+1; i++ {
+		select {
+		case checksumInfo := <-checksumInfoCh:
+			if checksumInfo.err != nil {
+				// only need to return the first error, others are context cancel error
+				if firstErr == nil {
+					firstErr = checksumInfo.err
+					cancel1()
+				}
+
+				continue
```

Comment:
use `continue` so we can get all results from every goroutine, this will be very fast because other goroutines will exit with `context canceled` error